test/unit/scan/test_scan_policies.py

```diff
@@ -7,14 +7,207 @@
+        reason="Symlinks not supported on PyPy/Windows"
+    )
+    def test_folder_with_symlink_to_file(self, tmp_path):
+        d = tmp_path / "dir"
+        d.mkdir()
+
+        file = d / "file.txt"
+        file.write_text("content")
+
+        symlink_file = tmp_path / "symlink_file.txt"
+        symlink_file.symlink_to(file)
+
+        folder = LocalFolder(str(tmp_path))
+        local_paths = folder.all_files(reporter=MagicMock())
+
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert len(absolute_paths) == 2
+        assert absolute_paths == [
+            fix_windows_path_limit(str(file)),
+            fix_windows_path_limit(str(symlink_file))
+        ]
+
+    #FIXME the following two tests could be combined to avoid code duplication
+    # but I decide to keep them separate for now to make it easier to debug
+
+    @pytest.mark.skipif(
+        platform.system() == 'Windows' and platform.python_implementation() == 'PyPy',
+        reason="Symlinks not supported on PyPy/Windows"
+    )
+    @pytest.mark.timeout(5)  # Set a 5-second timeout for this test
+    def test_folder_with_circular_symlink(self, tmp_path):
+        d = tmp_path / "dir"
+        d.mkdir()
+
+        (d / "file1.txt").write_text("content1")
+
+        symlink_dir = d / "symlink_dir"
+        symlink_dir.symlink_to(d)
+
+        folder = LocalFolder(str(tmp_path))
+
+        policies_manager = ScanPoliciesManager()
+        policies_manager.max_symlink_visits = 3
+
+        local_paths = folder.all_files(reporter=MagicMock(), policies_manager=policies_manager)
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert len(absolute_paths) == 4
+        assert absolute_paths == [
+            fix_windows_path_limit(str(d / "file1.txt")),
+            fix_windows_path_limit(str(d / "symlink_dir" / "file1.txt")),
+            fix_windows_path_limit(str(d / "symlink_dir" / "symlink_dir" / "file1.txt")),
+            fix_windows_path_limit(
+                str(d / "symlink_dir" / "symlink_dir" / "symlink_dir" / "file1.txt")
+            ),
+        ]
+
+    @pytest.mark.skipif(
+        platform.system() == 'Windows' and platform.python_implementation() == 'PyPy',
+        reason="Symlinks not supported on PyPy/Windows"
+    )
+    @pytest.mark.timeout(5)  # Set a 5-second timeout for this test
+    def test_circular_symlink_with_limit_1(self, tmp_path):
+        d = tmp_path / "dir"
+        d.mkdir()
+
+        (d / "file1.txt").write_text("content1")
+
+        symlink_dir = d / "symlink_dir"
+        symlink_dir.symlink_to(d)
+
+        policies_manager = ScanPoliciesManager()
+        policies_manager.max_symlink_visits = 1
+
+        folder = LocalFolder(str(tmp_path))
+        local_paths = folder.all_files(reporter=MagicMock(), policies_manager=policies_manager)
+        absolute_paths = [path.absolute_path for path in list(local_paths)]
+
+        assert len(absolute_paths) == 2
```

Comment:
```suggestion
        assert len(absolute_paths) == 2
```
the assert below already checks list length. Moreover if this assert fails, you'll get something like `3 != 2` which is way less useful than the feedback from the assert below.